src/js/actions/layers.js

```diff
@@ -745,10 +767,6 @@ define(function (require, exports) {
             layers = Immutable.List.of(layers);
         }
 
-        if (descendants) {
```

Comment:
Just moved this block down so that it came after the filtering step, which could cause a short circuit.